vespajlib/src/main/java/com/yahoo/text/Text.java

```diff
@@ -91,35 +91,59 @@ public static boolean isTextCharacter(int codepoint) {
      * returns the first illegal code point if one is found.
      */
     public static OptionalInt validateTextString(String string) {
-        for (int i = 0; i < string.length(); i++) {
+        for (int i = 0; i < string.length(); ) {
             int codePoint = string.codePointAt(i);
             if ( ! Text.isTextCharacter(codePoint))
                 return OptionalInt.of(codePoint);
 
-            if (Character.isHighSurrogate(string.charAt(i)))
-                ++i; // // codePointAt() consumes one more char in this case
+            int charCount = Character.charCount(codePoint);
+            if (Character.isHighSurrogate(string.charAt(i))) {
+                if ( charCount == 1) {
+                    return OptionalInt.of(string.codePointAt(i));
+                } else if ( !Character.isLowSurrogate(string.charAt(i+1))) {
+                    return OptionalInt.of(string.codePointAt(i+1));
```

Comment:
`charAt(i+1)` may throw `IndexOutOfBoundsException` if high surrogate is last char in string. Also check if i+1 would go out of bounds if `charCount` is 2?

Reply:
charAt(i+1) will not be reached as it will return at the charCount == 1 branch.
If charCount is 2 both i and i+1 will be exist.